src/components/WorkBuddies/MyWorkBuddy.tsx

```diff
@@ -0,0 +1,118 @@
+import { Fragment, useContext, useEffect, useState } from "react";
+import { BuddiesContext } from "../BuddiesContext";
+import classNames from "classnames";
+import "./work-buddies.css";
+import { useReducer } from "react";
+import { Buddy } from "./BuddyInterface";
+import { memo } from "react";
+
+interface propsInterface {
+  myBuddy: Buddy;
+}
+
+const reducer = (
+  index: number,
+  action: { type: string; contextData?: Buddy[]; myBuddyIndex?: number }
+) => {
+  switch (action.type) {
+    case "next":
+      if (action.contextData) {
+        if (index + 1 < action.contextData.length) return index + 1;
+        else return 0;
+      } else return 0;
+
+    case "prev":
+      if (action.contextData) {
+        if (index - 1 > -1) return index - 1;
+        else return action.contextData.length - 1;
+      } else return 0;
+
+    case "my-buddy":
+      if (action.myBuddyIndex) {
+        return action.myBuddyIndex;
+      } else return 0;
+
+    default:
+      return 1;
+  }
+};
+
+const MyWorkBuddy = memo(({ myBuddy }: propsInterface) => {
```

Comment:
![image](https://github.com/TheaAleksovDev/ReactConceptsPractice/assets/26555796/b37e42a8-4906-4f87-92bc-6d2e90eab1e2)

You may need to take the individual props of the `myBuddy` object

Objects can have different references even if they have the same properties.
The consequence of that is **memo** becomes useless.

If you want `memo` to actually work, you can expect primitives:
    - name: string;
    - url: string;
    - isMine?: boolean;

Reply:
If you need to pass objects, React allows you to pass a custom comparison function:
- ![image](https://github.com/TheaAleksovDev/ReactConceptsPractice/assets/26555796/b10efa5d-0577-4bb1-b4a0-5a06cd8e6e4b)

Reply:
Got it. I will pass buddy's name, then conditionally change color of the name if it's mine, and that way I'll actually use it in this component and resolve comments that you made about it.